spyder_kernels/utils/iofuncs.py

```diff
@@ -288,58 +289,101 @@ def save_dictionary(data, filename):
             arr_fname = osp.splitext(filename)[0]
             for name in list(data.keys()):
-                if isinstance(data[name], np.ndarray) and data[name].size > 0:
-                    # Saving arrays at data root
-                    fname = __save_array(data[name], arr_fname,
-                                       len(saved_arrays))
-                    saved_arrays[(name, None)] = osp.basename(fname)
-                    data.pop(name)
-                elif isinstance(data[name], (list, dict)):
-                    # Saving arrays nested in lists or dictionaries
-                    if isinstance(data[name], list):
-                        iterator = enumerate(data[name])
-                    else:
-                        iterator = iter(list(data[name].items()))
-                    to_remove = []
-                    for index, value in iterator:
-                        if isinstance(value, np.ndarray) and value.size > 0:
-                            fname = __save_array(value, arr_fname,
-                                               len(saved_arrays))
-                            saved_arrays[(name, index)] = osp.basename(fname)
-                            to_remove.append(index)
-                    for index in sorted(to_remove, reverse=True):
-                        data[name].pop(index)
+                try:
+                    if isinstance(data[name],
+                                  np.ndarray) and data[name].size > 0:
+                        # Save arrays at data root
+                        fname = __save_array(data[name], arr_fname,
+                                             len(saved_arrays))
+                        saved_arrays[(name, None)] = osp.basename(fname)
+                        data.pop(name)
+                    elif isinstance(data[name], (list, dict)):
+                        # Save arrays nested in lists or dictionaries
+                        if isinstance(data[name], list):
+                            iterator = enumerate(data[name])
+                        else:
+                            iterator = iter(list(data[name].items()))
+                        to_remove = []
+                        for index, value in iterator:
+                            if isinstance(value,
+                                          np.ndarray) and value.size > 0:
+                                fname = __save_array(value, arr_fname,
+                                                     len(saved_arrays))
+                                saved_arrays[(name, index)] = (
+                                    osp.basename(fname))
+                                to_remove.append(index)
+                        for index in sorted(to_remove, reverse=True):
+                            data[name].pop(index)
+                    # If an array can't be saved with numpy for some reason,
+                    # leave the object intact and try to save it normally.
```

Comment:
Shouldn't this comment be above the `pass` statement below? I think the code would be easier to read that way.

Reply:
Yeah, you're right. I actually noticed it late last night but it was 5 am and I was just too tired to do another ``rebase`` to fix it. Fixed now.